src/Storages/StorageReplicatedMergeTree.cpp

```diff
@@ -285,21 +285,32 @@ StorageReplicatedMergeTree::StorageReplicatedMergeTree(
     , replicated_fetches_throttler(std::make_shared<Throttler>(getSettings()->max_replicated_fetches_network_bandwidth, getContext()->getReplicatedFetchesThrottler()))
     , replicated_sends_throttler(std::make_shared<Throttler>(getSettings()->max_replicated_sends_network_bandwidth, getContext()->getReplicatedSendsThrottler()))
 {
+    /// We create and deactivate all tasks for consistency.
+    /// They all will be scheduled and activated by the restarting thread.
     queue_updating_task = getContext()->getSchedulePool().createTask(
```

Comment:
Can the task run before we deactivate it?
Can we create it in an already deactivated state?

Reply:
1. No, because it's the constructor of the object.
2. No, we just have `createTask` and by default `deactivated = false` https://github.com/ClickHouse/ClickHouse/blob/97690577672ad3ed215b5e2b38b871197747baf6/src/Core/BackgroundSchedulePool.h#L148. From my point of view it's convenient API.